meta/schemas/device.yaml

```diff
@@ -0,0 +1,183 @@
+$id: http://openwrt.org/schema/device.yaml
+$schema: http://yaml-schema.org/draft-07/schema
+additionalProperties: true
+title: Schema for an OpenWrt device
+type: object
+
+required:
+  - vendor
+  - model
+  - cpu
+  - memory_mb
+  - storage
+  - target
+
+examples:
+  - bootloader: RedBoot
+    cpu:
+      cores: 2
+      mhz: 500
+      model: slow_cpu_2
+    cpu: Atheros AR2315
+    cpu_cores: 1
+    cpu_mhz: 180
+    detachable_antennas: 1
+    device_type: WiFi Router
+    ethernet:
+      - speed_mbit: 100
+        ports: 1
+        label: lan
```

Comment:
To be taken into account: There are devices out there that have e.g. 5x Gbit + 1x 100Mbit, i.e. multiple values must be possible.

Reply:
That should work as this is an array. You could do the following:

```
ethernet:
  - speed_mbit: 100
    port_count: 1
    function: control
  - speed_mbit: 1000
    port_count: 5
    function: lan
```

Reply:
That was also my idea of how it should work. However, "control" is something defined by the vendor which actually is not implemented that way in OpenWrt?

Reply:
I just came up with control. I interpret *function* as something describe the functionality or the label attached to the ports.